### ISSUE TITLE
Empty Configuration Causes Issues with Pre-Check Hooks

### ISSUE THREAD

**Issue by crazy54** [sford/yet-another-commit-checker#126]
**_First off, our server is Stash v3.11.4 -  I know its old, not my fault :(_** 

The issue was discovered this morning when I had a few devs tell me they were not able to check code in, and were seeing the following message:

```
Commit e24a814b created locally in repository C:\xxx-main
Error encountered while pushing to the remote repository: Response status code does not indicate success: 504 (Gateway Time-out).
Commit 20d592cc created locally in repository C:\xxx-main
Commit c90bade3 created locally in repository C:\xxx-main
The following errors were reported during push:
       refs/heads/feature/CASE-8826-Modify-Pass-Rule-Filter-on-AJD-Data-Source-to-be-Multiselect, pre-receive hook declined
```

Based on this I figured a plugin was not happy or the user was the issue.  I tried to push a change to the repo myself and got the same error. I went on the server and put the instance into safe mode and this resolved the issue. I put the server back into normal use and began to disable a few plugins until my push worked  The error went away when I disabled YACC.  

YACC in our instance was not setup with anything - the config was blank and we never had an issue before this.  I was able to grab the following from the atlassian-stash.log which made me put the server into safe mode:

`2016-09-26 11:51:41,979 DEBUG [http-nio-7990-exec-1] john.doe @10N1ZH8x711x7085x0 c2r2l2 192.168.12.176,127.0.0.1 "GET /rest/api/latest/projects/xxxx/repos/services.reporting/pull-requests/103/activities HTTP/1.0" com.atlassian.stash.scm.BaseCommand Executed /usr/bin/git rev-list --format=%H%x02%P%x02%aN%x02%aE%x02%at%n%B%n%x03END%x04 --ignore-missing --no-walk --stdin --
`

For now I have left YACC disabled and will revisit when Dev decides to set it up a bit.

**Comment by sford** [comment]
Without looking at this in too much detail, this sounds like #92, which was fixed in YACC 1.11.

There was a problem with global setting support where YACC was examining commits even if it wasn't configured. This caused problems with large repositories.

Unfortunately, YACC 1.11 requires Bitbucket Server 4.0, which doesn't help you. Sorry :-(
